onnxconverter_common/float16.py

```diff
@@ -82,7 +82,34 @@ def make_value_info_from_tensor(tensor):
                          'FeatureVectorizer', 'Imputer', 'LabelEncoder', 'LinearClassifier', 'LinearRegressor',
                          'Normalizer', 'OneHotEncoder', 'SVMClassifier', 'SVMRegressor', 'Scaler',
                          'TreeEnsembleClassifier', 'TreeEnsembleRegressor', 'ZipMap', 'NonMaxSuppression', 'TopK',
-                         'RoiAlign', 'Resize', 'Range', 'CumSum', 'Min', 'Max', 'Upsample']
+                         'RoiAlign', 'Range', 'CumSum', 'Min', 'Max', 'Upsample']
+
+
+# find all the constant input for specified op_type and index of input
+KEEP_ORIGINAL_DATA_TYPE_LIST = {
+    "Resize": 2,    # usage: the key:value means Resize operator, input[2]
+}
+
+
+# find all the constant input for specified op_type
+def get_nodes_to_keep_data_type(model):
+    inputs_constant = []
+    for op_type, input_index in KEEP_ORIGINAL_DATA_TYPE_LIST.items():
+        # find all the constant input[index] of operator by specified op_type
+        for node in model.graph.node:
+            if (node.op_type == op_type):
+                if len(node.input) <= input_index:
+                    raise ValueError('The input_index is larger than the total inputs count: %s' % op_type)
+                prev_node_output_name = node.input[input_index]
+                inputs_constant.append(prev_node_output_name)
+
+    # go through graph again to get all constant nodes by name
+    constant_input_nodes = []
+    for node in model.graph.node:
```

Comment:
nit: Probably we can add a judgement here that if inputs_constant is empty, return [] directly to avoid going through all of nodes in the graph without doing anything.

Reply:
good idea